### PR TITLE
On-Screen notification > WP admin settings

### LINKED ISSUE #2482: On-Screen Notifications > WP-Admin Settings
### Scope

**Design**
https://sketch.cloud/s/5d225452-8a1d-4f2c-95da-0eb927818399

**Instruction Video**
https://www.loom.com/share/2eb07e341fef421c84de6ad12b47cdb4

Settings

On-Screen Notifications
- Enable on-screen notifications if checked

Position on Screen
- Set whether notifications are aligned to the bottom left or bttom right of screen

Mobile Support
- Select whether to show notifications on screens smaller than 500px.

Automatically Hide
- Select whether to automaticalyl hide notifications
- Use the options: Never Hide, After 1 minute, After 3 Minutes, After 5 Minutes, After 15 Minutes

Show in Browser Tab
- See here: https://github.com/buddyboss/buddyboss-platform/issues/2490

### QA Needed
Test cases to be written and posted here by the QA team. (@joshuamarkt)

### Associated Issues
https://github.com/buddyboss/buddyboss-platform/issues/2483

### ProductBoard URL
https://buddyboss.productboard.com/feature-board/2860318-features-in-progress/features/7678066/detail

---

**On-Screen Notifications Issues**
https://github.com/buddyboss/buddyboss-platform/issues/2482
https://github.com/buddyboss/buddyboss-platform/issues/2490
https://github.com/buddyboss/buddyboss-platform/issues/2483

### LINKED ISSUE #2483: On-Screen Notifications
### Scope

**Design**
https://sketch.cloud/s/5d225452-8a1d-4f2c-95da-0eb927818399

**Instruction Video**
https://www.loom.com/share/2eb07e341fef421c84de6ad12b47cdb4

On-screen notifications show users new notifications received while they are on a page.

This is the first implementation of the feature using AJAX. We will extend it in the future by connecting it to a socket service for instant notifications without the need for continuous AJAX actions.

After the page loads, start running a check for new notifications every 30 seconds. 

Retrieve notifications that meet the following conditions:
- They are NOT marked as read
- They have been created in the previous 30 seconds

Show the retrieved notifications on the screen, matching the design in the Sketch file.

Notifications are ordered from most to least recent, top to bottom.

If there is more than one notification of the same type found, use the “multiple” format. (@sourabhmatolia to confirm if we can use this functionality)

Every 30 seconds, load new notifications created since the last check. If new notifications are found, remove the orange border from previously loaded notifications.

When a notification is closed, it swipes out to the side. The notifications underneath move up.

When there is more than one notification, the Close button shows.

When there are more than three notifications, show a hidden notification under the third notification.

Change the background color of the notification on hover.

Match the position of the notifications on the screen to the "Position on Screen" setting in the WP-Admin.

On screens smaller than 500px, use the small screen design. Only one notification every shows in this style. 
Disable notifications on small screens if "Mobile Support" is deactivated in the WP-Admin.

If the "Automatically Hide" settings is used in the WP-Admin, then remove notifications after the time set in the setting.

### QA Needed
Test cases to be written and posted here by the QA team. (@joshuamarkt)

### Associated Issues
https://github.com/buddyboss/buddyboss-platform/issues/2482

### ProductBoard URL
https://buddyboss.productboard.com/feature-board/2860318-features-in-progress/features/7678066/detail

---

**On-Screen Notifications Issues**
https://github.com/buddyboss/buddyboss-platform/issues/2482
https://github.com/buddyboss/buddyboss-platform/issues/2490
https://github.com/buddyboss/buddyboss-platform/issues/2483

### PR DESCRIPTION
**Pull request by asaquzzamanbb**
### All Submissions:

* [ ] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [ ] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2482
Fixes #2483

### Proof Screenshots or Video
https://www.loom.com/share/a5692338ff64400eb5c54ef3bab73929

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Comment by mikejyoung** [comment]
@amincdev @shashank-buddyboss @asaquzzamanbb 

Some changes to On-Screen Notifications settings:
https://www.loom.com/share/e98a74a734314c40a80d3398b7f9f268

CC: @sourabhmatolia

**Comment by shashank-buddyboss** [comment]
@mikejyoung 

Changes (https://github.com/buddyboss/buddyboss-platform/pull/2519#issuecomment-869136966) are done > https://prnt.sc/177nufg

Thanks

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@mahedihasannoman all the newly functions need the `bb` prefix instead of `bp`

File: src/bp-templates/bp-nouveau/includes/notifications/functions.php [hunk 1]
```diff
@@ -245,3 +245,84 @@ function bp_nouveau_notifications_delete_link( $link = '' ) {
 		'dashicons-trash'
 	);
 }
+
+function bb_notification_avatar() {
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/notifications/functions.php:249]: @asaquzzamanbb need a function doc block

File: src/bp-templates/bp-nouveau/includes/notifications/functions.php [hunk 2]
```diff
@@ -245,3 +245,84 @@ function bp_nouveau_notifications_delete_link( $link = '' ) {
...
+				)
+			);
+			?>
+			<?php ( isset( $user ) ? bb_current_user_status( $user->ID ) : '' ); ?>
+		</a>
+		<?php
+	}
+}
+
+function bb_current_user_status( $user_id ) {
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/notifications/functions.php:307]: @asaquzzamanbb need doc block

**Review by chetansatasiya — APPROVED** [review]
@asaquzzamanbb LGTM

**Pull request merged**